headers/data/overlay29.h

```diff
@@ -102,5 +102,6 @@ extern struct tile DEFAULT_TILE;
 extern bool HIDDEN_STAIRS_SPAWN_BLOCKED;
 extern void* FIXED_ROOM_DATA_PTR;
 extern struct dungeon_fades* DUNGEON_FADES_PTR;
+extern uint32_t POSITIONS_COUNTER;
```

Comment:
For a global variable, this name is sort of generic. Can we qualify it a little more? How about `AI_THROWN_ITEM_TARGET_POSITIONS_COUNT`?

Reply:
I actually discovered that this counter is also used in FUN_0230faa0 for multiple arrays (one being direction enums and another being probabilities). I don't know what non-generic name could encompass all of these cases, do you have any ideas?

Reply:
It looks like `FUN_0230f388` is a higher-level function for deciding...something, and it calls into (and is the only caller of) both `FUN_0230f824` and `GetPossibleAiArcItemTargets`. Additionally, the first call site of the subfunctions within the higher-level one is inside the block that checks `MonsterCanThrowItems`. So, I think `FUN_0230f824` must also be related to thrown items. Based on what you said about it dealing with directions and probabilities, perhaps it's a `THROWN_LINE` counterpart to the `THROWN_ARC` function you found. Additionally, this `POSITIONS_COUNTER` value is only accessed by the aforementioned three functions.

So, to me it seems like maybe it counts the number of possible actions the AI can take with a thrown item in a given category. With arc-thrown items, that can be encoded by a list of positions of targets. With line-thrown items, it seems plausible that that might be encoded by a list of directions to throw, and the corresponding probability of throwing in each direction? So maybe something like `AI_THROWN_ITEM_ACTION_CHOICE_COUNT`. We can add what we know in the description for the symbol.

Reply:
All of this is consistent with Rescue Team.

`FUN_0230f824` ([FindStraightThrowableItems](https://github.com/pret/pmd-red/blob/master/src/dungeon_ai_items.c#L274) in RT) is the `THROWN_LINE` counterpart to `GetPossibleAiArcItemTargets`.

`FUN_0230f388` ([DecideUseItem](https://github.com/pret/pmd-red/blob/master/src/dungeon_ai_items.c#L64) in RT) is the top-level function for deciding if an AI should use its held item and where to throw it (if applicable).

`POSITIONS_COUNTER` ([gNumPotentialTargets](https://github.com/pret/pmd-red/blob/master/src/dungeon_ai_items.c#L37) in RT) is used by the thrown line/arc functions to count the number of targets the AI has found for throwing its item. Line-thrown items are tracked with two parallel arrays of direction + probability, while arc-thrown items track the positions of targets.

Reply:
I also made these findings yesterday. I've added them, this should be everything related to AI item use.